common/build/eslint-config-fluid/src/custom-rules/no-restricted-tags-imports.js

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const path = require("path");
+const { Project } = require("ts-morph");
+
+/**
+ * Validates if a tag is correctly formatted.
+ * @param tag - The tag to be validated.
+ * @param context - The context object provided by ESLint.
+ * @returns true if valid tag, else false.
+ */
+function isTagValid(tag, context) {
+	if (!tag.startsWith("@")) {
+		context.report({
+			loc: { line: 1, column: 0 },
+			message: `Invalid tag format in rule configuration: '{${tag}}'. Tags should start with '@'.`,
+			data: { tag },
+		});
+	} else {
+		return true;
+	}
+}
+
+/**
+ * Processes the tags array to ensure all tags are valid.
+ * @param tags - Array of tags to be processed.
+ * @param context - The context object provided by ESLint.
+ * @returns A set of validated tags.
+ */
+function processTags(tags, context) {
+	return new Set(tags.filter((tag) => isTagValid(tag, context)));
+}
+
+/**
+ * Validate the exceptions object, ensuring all tags and paths are valid.
+ * @param exceptions - The exceptions object from the rule configuration.
+ * @param context - The context object provided by ESLint.
+ * @returns - An object with tags as keys and Sets of paths as values.
+ */
+function processExceptions(exceptions, context) {
+	const processedExceptions = {};
+	Object.keys(exceptions).forEach((tag) => {
+		if (isTagValid(tag, context)) {
+			processedExceptions[tag] = new Set(exceptions[tag]);
+		}
+	});
+	return processedExceptions;
+}
+
+/**
+ * Resolves the path of an imported module to an absolute path.
+ *
+ * @param importPath - The path used in the import statement.
+ * @param currentFilePath - The absolute path of the current file.
+ * @return - The absolute path of the imported module.
+ */
+function resolveImportPath(importPath, currentFilePath) {
+	return path.isAbsolute(importPath)
+		? importPath
+		: path.resolve(path.dirname(currentFilePath), importPath);
+}
+
+module.exports = {
+	meta: {
+		type: "problem",
+		docs: {
+			description:
+				"This rule restricts imports from specified tags or non-public APIs. This to prevent accidental dependencies on internal, unstable or undocumented parts of the codebase.",
```

Comment:
I think the rule is a bit more general than this. I might say something like "Restricts imports based on one or more TSDoc tags they are annotated with. Can be used to enforce release tag rules against imports, etc."